### PR TITLE
DEV: add pnpm compat in preparation of new changes to theme-skeleton

### PR DESCRIPTION
**Pull request by tyb-talks**
We are preparing to switch to `pnpm` for all themes/plugins now that the change has landed in core. This change makes the discourse_theme CLI compatible with that change to theme-skeleton. We will remove all usage of yarn in an upcoming minor version bump once pnpm is fully rolled out.

For more context: https://meta.discourse.org/t/discourse-core-is-switching-to-pnpm-for-js-package-management/324521

### CODE REVIEW

File: test/fixtures/skeleton-lite/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (tyb-talks) [inline test/fixtures/skeleton-lite/package.json:1]: adding this file is necessary for tests to pass - yarn simply creates empty node_modules and lockfile, but pnpm install would raise an error here. In any case, we should expect the actual theme skeleton repo to always have at least a package.json file.

File: discourse_theme.gemspec [hunk 2]
```diff
@@ -36,7 +36,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "guard"
   spec.add_development_dependency "guard-minitest"
   spec.add_development_dependency "webmock"
-  spec.add_development_dependency "rubocop-discourse", "~> 3.6.0"
+  spec.add_development_dependency "rubocop-discourse", "~> 3.8.1"
```
Comment (tyb-talks) [inline discourse_theme.gemspec:39]: Update required to resolve an earlier error raised in linting CI step.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,13 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [2.1.1] - 2024-03-25
+## [2.1.3] - 2024-10-09
+
+### Added
+
+- Made `new` command compatible with the replacement of `yarn` with `pnpm` as a package manager, and will prompt users to install `pnpm` if not installed already.
+
+## [2.1.2] - 2024-03-25
```
Comment (cvx) [inline CHANGELOG.md]: a minor correction
```suggestion
## [2.1.2] - 2024-04-16
```

**Review by cvx — APPROVED** [review]
